src/main/java/org/opensearch/knn/indices/ModelMetadata.java

```diff
@@ -240,22 +260,31 @@ public int hashCode() {
     public static ModelMetadata fromString(String modelMetadataString) {
         String[] modelMetadataArray = modelMetadataString.split(DELIMITER, -1);
 
-        if (modelMetadataArray.length != 7) {
+        if (modelMetadataArray.length == 7) {
```

Comment:
Add thorough comment around these cases. Also, add log statement here for which one it is reading from.